modules/social_features/social_event/modules/social_event_an_enroll/src/Form/EventAnEnrollForm.php

```diff
@@ -89,64 +120,96 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       '#button_level' => 'raised',
       '#value' => $submit_text,
       '#disabled' => !$enrollment_open,
+      '#ajax' => [
+        'callback' => '::ajaxSubmitAnEnrollForm',
+        'disable-refocus' => TRUE,
+        'progress' => 'none',
+        'url' => Url::fromRoute('social_event_an_enroll.enroll_form', ['node' => $node->id()]),
+        'options' => [
+          'query' => [
+            FormBuilderInterface::AJAX_FORM_REQUEST => TRUE,
+          ],
+        ],
+      ],
     ];
 
     return $form;
   }
 
   /**
-   * {@inheritdoc}
+   * Rebuilds form after ajax submit.
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
    */
-  public function submitForm(array &$form, FormStateInterface $form_state) {
-    $current_user = $this->currentUser;
-    $uid = $current_user->id();
-
-    if ($current_user->isAnonymous()) {
-      $nid = $form_state->getValue('field_event');
-      $values = $form_state->getValues();
+  public function ajaxSubmitAnEnrollForm(array &$form, FormStateInterface $form_state): Response {
+    $response = new AjaxResponse();
+    $token = NULL;
+    $nid = $form_state->getValue('field_event');
+    $uid = $this->currentUser()->id();
+    $store = $this->tempStoreFactory->get('social_event_an_enroll');
+    $an_enrollments = $store->get('enrollments');
+    $values = $form_state->getValues();
+    /** @var \Drupal\node\NodeInterface $node */
+    $node = $this->entityTypeManager->getStorage('node')->load($nid);
+
+    // Check if there is enrollment with the same email.
+    $enrollments = $this->enrollmentStorage->loadByProperties([
+      'field_event' => $nid,
+      'field_email' => $form_state->getValue('field_email'),
+    ]);
+
+    if ($enrollment = array_pop($enrollments)) {
+      $token = $enrollment->get('field_token')->getString();
+      $response->addCommand(new CloseDialogCommand());
+      $response->addCommand(new MessageCommand(
+        $this->t('You have been already enrolled to this event. You have also received a notification via email.'),
+        NULL,
+        ['type' => 'status']
+      ));
+    }
+    else {
+      $token = Crypt::randomBytesBase64();
 
       $values['user_id'] = $uid;
       $values['field_account'] = $uid;
       $values['field_enrollment_status'] = '1';
+      $values['field_token'] = $token;
+
+      $this->enrollmentStorage->create($values)->save();
+      $enroll_confirmation = $this->entityTypeManager->getViewBuilder('node')->view($node, 'teaser');
+      $enroll_confirmation['#theme'] = 'event_enrollment_confirmation';
+      $response->addCommand(new OpenModalDialogCommand(
+        $this->t('Thanks for enrolling!'),
+        $enroll_confirmation,
+        [
+          'width' => '479px',
```

Comment:
Does this dialog works properly on mobile?

Reply:
<img width="920" alt="Screenshot 2022-04-11 at 14 17 13" src="https://user-images.githubusercontent.com/50984627/162728934-789eb69b-2662-4187-819c-a563cf617a66.png">
Seems it looks fine.